integration_tests/internal_payment_test.go

```diff
@@ -399,10 +399,9 @@ func (suite *PaymentTestSuite) TestInternalPaymentFail() {
 		fmt.Printf("Error when getting invoices %v\n", err.Error())
 	}
 
-	// check if first one is settled, but second one error (they are ordered desc by id)
-	assert.Equal(suite.T(), 2, len(invoices))
-	assert.Equal(suite.T(), common.InvoiceStateError, invoices[0].State)
-	assert.Equal(suite.T(), common.InvoiceStateSettled, invoices[1].State)
+	// check if first one is settled, second one is erroneous and hence isn't listed
+	assert.Equal(suite.T(), 1, len(invoices))
+	assert.Equal(suite.T(), common.InvoiceStateSettled, invoices[0].State)
```

Comment:
can we query for the second one differently and check that it has the state error?

Reply:
Fixed